hydra-node/src/Hydra/API/Server.hs

```diff
@@ -136,24 +140,31 @@ runAPIServer ::
 runAPIServer host port party tracer history callback responseChannel = do
   traceWith tracer (APIServerStarted port)
   handle onIOException $
-    runServer (show host) (fromIntegral port) $ \pending -> do
-      let path = requestPath $ pendingRequest pending
-      queryParams <- uriQuery <$> mkURIBs path
-      con <- acceptRequest pending
-      chan <- STM.atomically $ dupTChan responseChannel
-      traceWith tracer NewAPIConnection
-
-      -- api client can decide if they want to see the past history of server outputs
-      if shouldNotServeHistory queryParams
-        then forwardGreetingOnly con
-        else forwardHistory con
-
-      -- api client can decide if they want tx's to be displayed as CBOR instead of plain json
-      let txDisplay = decideOnTxDisplay queryParams
-
-      withPingThread con 30 (pure ()) $
-        race_ (receiveInputs con) (sendOutputs chan con txDisplay)
+    let serverSettings = setHost (ipToHostPreference host) $ setPort port $ defaultSettings
+    in runSettings serverSettings $ websocketsOr defaultConnectionOptions wsApp httpApp
  where
+  wsApp pending = do
+    let path = requestPath $ pendingRequest pending
+    queryParams <- uriQuery <$> mkURIBs path
+    con <- acceptRequest pending
+    chan <- STM.atomically $ dupTChan responseChannel
+    traceWith tracer NewAPIConnection
+
+    -- api client can decide if they want to see the past history of server outputs
+    if shouldNotServeHistory queryParams
+      then forwardGreetingOnly con
+      else forwardHistory con
+
+    -- api client can decide if they want tx's to be displayed as CBOR instead of plain json
+    let txDisplay = decideOnTxDisplay queryParams
+
+    withPingThread con 30 (pure ()) $
+      race_ (receiveInputs con) (sendOutputs chan con txDisplay)
+        `catch` onServerException con
+
+  httpApp _ respond =
+    respond $ responseLBS status400 [] "Currently Hydra supports only WebSocket requests."
```

Comment:
Could: remove "currently" and either call this `hydra-node` or just not put a subject

```suggestion
    respond $ responseLBS status400 [] "only WebSocket connections supported"
```